public/app/plugins/panel/heatmap/color_legend.ts

```diff
@@ -0,0 +1,300 @@
+///<reference path="../../../headers/common.d.ts" />
+import angular from 'angular';
+import _ from 'lodash';
+import $ from 'jquery';
+import d3 from 'd3';
+import {contextSrv} from 'app/core/core';
+import {tickStep} from 'app/core/utils/ticks';
+
+let module = angular.module('grafana.directives');
+
+/**
+ * Color legend for heatmap editor.
+ */
+module.directive('colorLegend', function() {
+  return {
+    restrict: 'E',
+    template: '<div class="heatmap-color-legend"><svg width="16.8rem" height="24px"></svg></div>',
```

Comment:
you can simplify the template a bit by using the color-legend element and add the class to this element instead, that way you do not need to find the 'heatmap-color-legend' element. 

```js
    template: '<svg width="16.8rem" height="24px"></svg>',
    link: function(scope, elem, attrs) {
      let ctrl = scope.ctrl;
      let panel = scope.ctrl.panel;

      elem.addClass('heatmap-color-legend');
```